errors.go

```diff
@@ -99,20 +99,30 @@ func (e *e) Cause() error {
 	return e.cause
 }
 
-// Wrap returns an error annotating the cause with message.
-// If cause is nil, Wrap returns nil.
-func Wrap(cause error, message string) error {
+func wrap(cause error, message string) error {
```

Comment:
Sorry for this getting dragged on a bit longer than it needed to. Copied Wrap to Wrapf and squashed into one commit.